dotcom-rendering/src/server/lib/logging.ts

```diff
@@ -25,16 +25,23 @@ const logFields = (logEvent: LoggingEvent): unknown => {
 	};
 	// log4js uses any[] to type data but we want to coerce it here
 	// because we now depend on the type to log the result properly
-	const data = logEvent.data[0] as unknown as
-		| Record<string, unknown>
-		| string;
+	const [data, additionalData] = logEvent.data;
+
+	if (typeof data === 'string' && typeof additionalData === 'object') {
```

Comment:
It's a little bit fiddly to work out what these conditions are satisfying. I think it's this?

- logging with a message
- logging with a message and data
- logging with no message and data

I wonder if naming things more descriptively could help?

```ts
const [arg1, arg2] = logEvent.data;
const message = typeof arg1 === 'string' ? arg1 : 'DCR Render event';
const data = typeof arg2 === 'object' ? arg2 : {};

return {
  message,
  ...coreFields,
  ...data,
}
```

Reply:
I was thinking of a way to make it look nice! This is great thanks!